hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -762,19 +773,21 @@ private void handleAutoResolvedTriggers() {
 
     private int checkMissingStates() {
         int numMatchingEvals = 0;
-        for (MissingState missingState : missingStates) {
+        Iterator<MissingState> itMissingStates = missingStates.iterator();
```

Comment:
Yes you are right.
The bytecode for each vs iterator is the same.
I first moved to an iterator, and when I discover the problem I protected the update with a synchronize in the other side.
Well, I would maintain it, at least for me it helps to remind that a collection managed by iterator can be updated outside of the loop concurrently.